### PR TITLE
[Remove Stream/Optional: Phase 4] LPS-182656 Remove Stream and optional usages for class: DDMRESTDataProvider

### PR DESCRIPTION
**Pull request by W2jx**
I have modified the code according to the [comment](https://github.com/liferay-core-infra/liferay-portal/pull/4407#issuecomment-1533510858).

Previous: https://github.com/liferay-core-infra/liferay-portal/pull/4407

### CODE REVIEW

**Comment by catarineaguiar** [comment]
ci:test:forms

**Comment by catarineaguiar** [comment]
ci:test:forms

**Comment by catarineaguiar** [comment]
no regressions found.

**Comment by W2jx** [comment]
no conflicts(5/8 15:34)

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-data-provider-impl/src/main/java/com/liferay/dynamic/data/mapping/data/provider/internal/rest/DDMRESTDataProvider.java [hunk 1]
```diff
@@ -520,21 +509,33 @@ private Map<String, String> _getPathInputParametersMap(
...
+		sb.append(StringPool.QUESTION);
+
+		for (Map.Entry<String, String> entry : entries) {
+			sb.append(entry.getKey());
+			sb.append(StringPool.EQUAL);
+			sb.append(entry.getValue());
+			sb.append(StringPool.AMPERSAND);
+		}
+
+		if (sb.index() > 4) {
```
Comment (dantewang) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-data-provider-impl/src/main/java/com/liferay/dynamic/data/mapping/data/provider/internal/rest/DDMRESTDataProvider.java:534]: @W2jx It's safer to check `if (!entires.isEmpty())`

**Review by dantewang — CHANGES_REQUESTED** [review]

**Comment by dantewang** [comment]
@W2jx Please resend as we talked.

**Pull request closed without merging**